bitcoin-units/Cargo.toml

```diff
@@ -0,0 +1,36 @@
+[package]
+name = "bitcoin-units"
+version = "0.1.0"
+authors = ["Andrew Poelstra <apoelstra@wpsoftware.net>", "The Rust Bitcoin developers"]
+license = "CC0-1.0"
+homepage = "https://github.com/rust-bitcoin/rust-bitcoin/"
+repository = "https://github.com/rust-bitcoin/rust-bitcoin/"
+documentation = "https://docs.rs/bitcoin/"
+description = "Bitcoin units"
+keywords = ["crypto", "bitcoin"]
+readme = "README.md"
+edition = "2018"
+
+# Please don't forget to add relevant features to docs.rs below.
+[features]
+default = ["std"]
+serde = ["actual-serde"]
+
+# Either "std" or "alloc" must be enabled.
+std = []
+alloc = ["core2/alloc"]
+
+[package.metadata.docs.rs]
+features = ["std"]
+rustdoc-args = ["--cfg", "docsrs"]
+
+[dependencies]
+core2 = { version = "0.3.0", default-features = false, optional = true }
```

Comment:
I don't think this will be needed, consensus encoding will be implemented once we move consensus encoding to a separate crate but we should just reexport `io` there and use that.

Reply:
Don't we need `core2` for `String`? Removing `String` requires a bit of work changing the error types. I thought to do that after creating the crate and before releasing it, would you prefer to see that done as part of this PR?

Reply:
No, we need `alloc` for `String`. So you should be able to remove this dependency right away.

For error types I'd prefer the approach I used in `ln-types`: if `alloc` is off the inputs will not be carried in errors and not printed. I have some macros for that there as well, feel free to copy whatever you need. I don't mind doing it in a subsequent PR.

Reply:
Cool, got it. Will have a go, cheers.

Reply:
Ah, removing `core2` is trivial, it should never have been there as a dependency as its unused - my fail. Fixing the new `bitcoin-units` crate so it can be built without an allocator is non-trivial, I think the `bitcoin-internals` crate should be created first so we can use macros from it and not re-create them all (e.g., `write_err`). I'll put this as draft and work on `bitcoin-internals`.

Reply:
This is WIP in #1297 I made that draft because nothing uses it. Maybe try rebasing this one on top of that and use the macros. You'll need to rename the error type and redesign parsing a bit.

Reply:
Awesome, can do!

Reply:
> You'll need to rename the error type and redesign parsing a bit.

I only messed with the errors and the unit tests not with how parsing is done.